### PR TITLE
Context Manager to modify config.

### PR DESCRIPTION
**Pull request by Carreau**
Test will fail as `_validate_config`, actually pop values from config.

### CODE REVIEW

**Comment by takluyver** [comment]
What's the use case you want this for?

If we're doing programmatic modification of config files, I'd like to look into round-tripping parsers (e.g. the [iniparse](https://pypi.python.org/pypi/iniparse) module, but preferably something that's still active). With the standard library configparser module, any formatting or comments you have in the config file is lost when you overwrite it. And comments are one of the main reasons to use ini instead of JSON.

**Comment by Carreau** [comment]
> If we're doing programmatic modification of config files,

Yes, programatic modification.

Would you be ok in just splitting the validate logic out, so that I can use a rough programatic way to modify ?


We can also have that in flit (private) and have round-tripping as a later feature.

**Comment by takluyver** [comment]
You mean splitting validate into a separate function from reading
configuration? Sure, I don't see a problem with that.

**Comment by Carreau** [comment]
> You mean splitting validate into a separate function from reading
> configuration? Sure, I don't see a problem with that.

Done, removed the context manager and change the validate method not to modify the config object it got as parameter.

**Comment by takluyver** [comment]
Thanks. At the airport now, but I'll try to have a look this evening from
the hotel.

**Comment by Carreau** [comment]
Ping BTW.

File: flit/inifile.py [hunk 1]
```diff
@@ -104,10 +104,22 @@ def verify_classifiers(classifiers):
...
     cp = configparser.ConfigParser()
     with path.open() as f:
         cp.read_file(f)
 
+    return cp
+
+
+def _validate_config(cp, path):
+    """
+    validate a config section and return a strip down version fro Metadata
```
Comment (takluyver) [inline flit/inifile.py]: * 'a config section' -> config (it's not a section, it's the whole thing, but there isn't a good word for that)
* It's not exactly stripped down - the config format includes some extra things under `[metadata]` that aren't strictly part of the distribution metadata. And it returns a dictionary containing several things, only one of which is metadata.

File: flit/inifile.py [hunk 2]
```diff
@@ -104,10 +104,22 @@ def verify_classifiers(classifiers):
 def read_pkg_ini(path):
     """Read and check the -pkg.ini file with data about the package.
```
Comment (takluyver) [inline flit/inifile.py]: Nothing to do with this PR, but we should update this docstring to say flit.ini

File: flit/inifile.py (hunk 1 above)
Comment (Carreau) [inline flit/inifile.py]: How's 

```
Validate a config and return a dict containing `module`,`metadata`,`script`,`entry_point` keys
```

File: flit/inifile.py (hunk 2 above)
Comment (Carreau) [inline flit/inifile.py]: Done.

**Comment by takluyver** [comment]
Thanks

**Pull request merged**